jetty-io/src/main/java/org/eclipse/jetty/io/ArrayRetainableByteBufferPool.java

```diff
@@ -430,19 +433,23 @@ public void dump(Appendable out, String indent) throws IOException
 
     private Pool<RetainableByteBuffer>.Entry findOldestEntry(long now, Pool<RetainableByteBuffer> bucket)
     {
+        // This method may be in the hot path, do not use Java streams.
+
         RetainedBucket.Entry oldestEntry = null;
+        RetainableByteBuffer oldestBuffer = null;
+        long oldestAge = 0;
         for (RetainedBucket.Entry entry : bucket.values())
         {
-            if (oldestEntry != null)
-            {
-                long entryAge = NanoTime.elapsed(entry.getPooled().getLastUpdate(), now);
-                if (entryAge > NanoTime.elapsed(oldestEntry.getPooled().getLastUpdate(), now))
-                    oldestEntry = entry;
-            }
-            else
-            {
-                oldestEntry = entry;
-            }
+            RetainableByteBuffer buffer = entry.getPooled();
+            // Concurrently removed? Try next.
+            if (buffer == null)
+                continue;
```

Comment:
I think returning here is probably better as the "oldest" calculation may now be wrong anyway after the concurrent access.
```suggestion
                return null;
```

Reply:
The comment is wrong, it is not a concurrent remove, but a case in between `reserve()` and `enable()`, so it's a very recent entry, so we still want to find the oldest.
Eventually, any entry that is found will have to go through an atomic `remove()`.

Reply:
OK... I still think that return null would be better.... but no harm continuing either.